ZenPacks/zenoss/LinuxMonitor/modeler/plugins/zenoss/cmd/linux/rpm.py

```diff
@@ -73,14 +111,29 @@ def parseResults(results):
                     gmtime(int(fields["INSTALLTIME"]))))
     return softwareDicts
 
+def parseResults(results):
+    softwareDicts = []
+    rpmResults, dpkgResults = results.split('_SPLIT_',1)
+    softwareDicts = parseRpmResults(rpmResults, softwareDicts)
+    softwareDicts = parseDpkgResults(dpkgResults, softwareDicts)
+    return softwareDicts
 
 class rpm(SoftwareCommandPlugin):
     """
-    Parse rpm -qa command output to get installed software.
+    Parse rpm -qa and dpkg-query command output to get installed software.
     """
 
-    command = r'`which rpm || echo true` -qa --qf "%s\n"' % QUERY_FORMAT
-
+    command = ( 'export PATH=$PATH:/bin:/sbin:/usr/bin:/usr/sbin; '
+                '`which rpm || echo true` -qa --qf "%{NAME}__FIELD__%{VERSION}__FIELD__%{RELEASE}__FIELD__%{INSTALLTIME}__FIELD__%{VENDOR}__FIELD__%{SUMMARY}\\n"; '
+                'echo "_SPLIT_"; '
+                'if command -v dpkg dpkg >/dev/null 2>&1; then '
```

Comment:
Some redhat/centos system can install dpkg on them, so this may give you unwanted results.
Also, why do you need to extecute dpkg twice to test if it exists. A single version should give you correct answer.
Note that Debian Ubuntu can also install RPM, giving you unwanted results here.